ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx

```diff
@@ -560,6 +552,9 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
         }
 
         const node = this.props.nodes[id];
+        if (!node) {
```

Comment:
fixes crashing issue

Reply:
I don't think this will conflict looking at #4028?

Reply:
This code will be unreachable after #4028. Also this fix is not exhaustive as there are other places in the file where null nodes may be returned under the right conditions. #4028 fixes them exhaustively.

Reply:
deleted